// upstox-js-sdk/dist/model/GetExpiredFuturesContractResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.GetExpiredFuturesContractResponse = void 0;
var _ApiClient = require("../ApiClient");
var _ExpiredFutureData = require("./ExpiredFutureData");
/*
 * OpenAPI definition
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 3.0.66
 *
 * Do not edit the class manually.
 *
 */

/**
 * The GetExpiredFuturesContractResponse model module.
 * @module model/GetExpiredFuturesContractResponse
 * @version v0
 */
class GetExpiredFuturesContractResponse {
  /**
   * Constructs a new <code>GetExpiredFuturesContractResponse</code>.
   * @alias module:model/GetExpiredFuturesContractResponse
   * @class
   */
  constructor() {}

  /**
   * Constructs a <code>GetExpiredFuturesContractResponse</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/GetExpiredFuturesContractResponse} obj Optional instance to populate.
   * @return {module:model/GetExpiredFuturesContractResponse} The populated <code>GetExpiredFuturesContractResponse</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new GetExpiredFuturesContractResponse();
      if (data.hasOwnProperty('status')) obj.status = _ApiClient.ApiClient.convertToType(data['status'], 'String');
      if (data.hasOwnProperty('data')) obj.data = _ApiClient.ApiClient.convertToType(data['data'], [_ExpiredFutureData.ExpiredFutureData]);
    }
    return obj;
  }
}

/**
 * Allowed values for the <code>status</code> property.
 * @enum {String}
 * @readonly
 */
exports.GetExpiredFuturesContractResponse = GetExpiredFuturesContractResponse;
GetExpiredFuturesContractResponse.StatusEnum = {
  /**
   * value: "success"
   * @const
   */
  success: "success",
  /**
   * value: "error"
   * @const
   */
  error: "error",
  /**
   * value: "partial_success"
   * @const
   */
  partialSuccess: "partial_success"
};
/**
 * @member {module:model/GetExpiredFuturesContractResponse.StatusEnum} status
 */
GetExpiredFuturesContractResponse.prototype.status = undefined;

/**
 * @member {Array.<module:model/ExpiredFutureData>} data
 */
GetExpiredFuturesContractResponse.prototype.data = undefined;
