// upstox-js-sdk/dist/model/GetMarketQuoteOptionGreekResponseV3.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.GetMarketQuoteOptionGreekResponseV3 = void 0;
var _ApiClient = require("../ApiClient");
var _MarketQuoteOptionGreekV = require("./MarketQuoteOptionGreekV3");
/*
 * OpenAPI definition
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 3.0.66
 *
 * Do not edit the class manually.
 *
 */

/**
 * The GetMarketQuoteOptionGreekResponseV3 model module.
 * @module model/GetMarketQuoteOptionGreekResponseV3
 * @version v0
 */
class GetMarketQuoteOptionGreekResponseV3 {
  /**
   * Constructs a new <code>GetMarketQuoteOptionGreekResponseV3</code>.
   * @alias module:model/GetMarketQuoteOptionGreekResponseV3
   * @class
   */
  constructor() {}

  /**
   * Constructs a <code>GetMarketQuoteOptionGreekResponseV3</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/GetMarketQuoteOptionGreekResponseV3} obj Optional instance to populate.
   * @return {module:model/GetMarketQuoteOptionGreekResponseV3} The populated <code>GetMarketQuoteOptionGreekResponseV3</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new GetMarketQuoteOptionGreekResponseV3();
      if (data.hasOwnProperty('status')) obj.status = _ApiClient.ApiClient.convertToType(data['status'], 'String');
      if (data.hasOwnProperty('data')) obj.data = _ApiClient.ApiClient.convertToType(data['data'], {
        'String': _MarketQuoteOptionGreekV.MarketQuoteOptionGreekV3
      });
    }
    return obj;
  }
}

/**
 * Allowed values for the <code>status</code> property.
 * @enum {String}
 * @readonly
 */
exports.GetMarketQuoteOptionGreekResponseV3 = GetMarketQuoteOptionGreekResponseV3;
GetMarketQuoteOptionGreekResponseV3.StatusEnum = {
  /**
   * value: "success"
   * @const
   */
  success: "success",
  /**
   * value: "error"
   * @const
   */
  error: "error",
  /**
   * value: "partial_success"
   * @const
   */
  partialSuccess: "partial_success"
};
/**
 * @member {module:model/GetMarketQuoteOptionGreekResponseV3.StatusEnum} status
 */
GetMarketQuoteOptionGreekResponseV3.prototype.status = undefined;

/**
 * @member {Object.<String, module:model/MarketQuoteOptionGreekV3>} data
 */
GetMarketQuoteOptionGreekResponseV3.prototype.data = undefined;
