// upstox-js-sdk/dist/model/GttOrderDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.GttOrderDetails = void 0;
var _ApiClient = require("../ApiClient");
var _Rule = require("./Rule");
/*
 * OpenAPI definition
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 3.0.66
 *
 * Do not edit the class manually.
 *
 */

/**
 * The GttOrderDetails model module.
 * @module model/GttOrderDetails
 * @version v0
 */
class GttOrderDetails {
  /**
   * Constructs a new <code>GttOrderDetails</code>.
   * Response data for order details
   * @alias module:model/GttOrderDetails
   * @class
   */
  constructor() {}

  /**
   * Constructs a <code>GttOrderDetails</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/GttOrderDetails} obj Optional instance to populate.
   * @return {module:model/GttOrderDetails} The populated <code>GttOrderDetails</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new GttOrderDetails();
      if (data.hasOwnProperty('type')) obj.type = _ApiClient.ApiClient.convertToType(data['type'], 'String');
      if (data.hasOwnProperty('exchange')) obj.exchange = _ApiClient.ApiClient.convertToType(data['exchange'], 'String');
      if (data.hasOwnProperty('quantity')) obj.quantity = _ApiClient.ApiClient.convertToType(data['quantity'], 'Number');
      if (data.hasOwnProperty('product')) obj.product = _ApiClient.ApiClient.convertToType(data['product'], 'String');
      if (data.hasOwnProperty('rules')) obj.rules = _ApiClient.ApiClient.convertToType(data['rules'], [_Rule.Rule]);
      if (data.hasOwnProperty('trading_symbol')) obj.tradingSymbol = _ApiClient.ApiClient.convertToType(data['trading_symbol'], 'String');
      if (data.hasOwnProperty('instrument_token')) obj.instrumentToken = _ApiClient.ApiClient.convertToType(data['instrument_token'], 'String');
      if (data.hasOwnProperty('gtt_order_id')) obj.gttOrderId = _ApiClient.ApiClient.convertToType(data['gtt_order_id'], 'String');
      if (data.hasOwnProperty('expires_at')) obj.expiresAt = _ApiClient.ApiClient.convertToType(data['expires_at'], 'Number');
      if (data.hasOwnProperty('created_at')) obj.createdAt = _ApiClient.ApiClient.convertToType(data['created_at'], 'Number');
    }
    return obj;
  }
}

/**
 * @member {String} type
 */
exports.GttOrderDetails = GttOrderDetails;
GttOrderDetails.prototype.type = undefined;

/**
 * @member {String} exchange
 */
GttOrderDetails.prototype.exchange = undefined;

/**
 * @member {Number} quantity
 */
GttOrderDetails.prototype.quantity = undefined;

/**
 * @member {String} product
 */
GttOrderDetails.prototype.product = undefined;

/**
 * @member {Array.<module:model/Rule>} rules
 */
GttOrderDetails.prototype.rules = undefined;

/**
 * @member {String} tradingSymbol
 */
GttOrderDetails.prototype.tradingSymbol = undefined;

/**
 * @member {String} instrumentToken
 */
GttOrderDetails.prototype.instrumentToken = undefined;

/**
 * @member {String} gttOrderId
 */
GttOrderDetails.prototype.gttOrderId = undefined;

/**
 * @member {Number} expiresAt
 */
GttOrderDetails.prototype.expiresAt = undefined;

/**
 * @member {Number} createdAt
 */
GttOrderDetails.prototype.createdAt = undefined;
