// upstox-js-sdk/dist/model/MarketQuoteSymbolLtpV3.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MarketQuoteSymbolLtpV3 = void 0;
var _ApiClient = require("../ApiClient");
/*
 * OpenAPI definition
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 3.0.66
 *
 * Do not edit the class manually.
 *
 */

/**
 * The MarketQuoteSymbolLtpV3 model module.
 * @module model/MarketQuoteSymbolLtpV3
 * @version v0
 */
class MarketQuoteSymbolLtpV3 {
  /**
   * Constructs a new <code>MarketQuoteSymbolLtpV3</code>.
   * @alias module:model/MarketQuoteSymbolLtpV3
   * @class
   */
  constructor() {}

  /**
   * Constructs a <code>MarketQuoteSymbolLtpV3</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/MarketQuoteSymbolLtpV3} obj Optional instance to populate.
   * @return {module:model/MarketQuoteSymbolLtpV3} The populated <code>MarketQuoteSymbolLtpV3</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new MarketQuoteSymbolLtpV3();
      if (data.hasOwnProperty('last_price')) obj.lastPrice = _ApiClient.ApiClient.convertToType(data['last_price'], 'Number');
      if (data.hasOwnProperty('instrument_token')) obj.instrumentToken = _ApiClient.ApiClient.convertToType(data['instrument_token'], 'String');
      if (data.hasOwnProperty('ltq')) obj.ltq = _ApiClient.ApiClient.convertToType(data['ltq'], 'Number');
      if (data.hasOwnProperty('volume')) obj.volume = _ApiClient.ApiClient.convertToType(data['volume'], 'Number');
      if (data.hasOwnProperty('cp')) obj.cp = _ApiClient.ApiClient.convertToType(data['cp'], 'Number');
    }
    return obj;
  }
}

/**
 * @member {Number} lastPrice
 */
exports.MarketQuoteSymbolLtpV3 = MarketQuoteSymbolLtpV3;
MarketQuoteSymbolLtpV3.prototype.lastPrice = undefined;

/**
 * @member {String} instrumentToken
 */
MarketQuoteSymbolLtpV3.prototype.instrumentToken = undefined;

/**
 * @member {Number} ltq
 */
MarketQuoteSymbolLtpV3.prototype.ltq = undefined;

/**
 * @member {Number} volume
 */
MarketQuoteSymbolLtpV3.prototype.volume = undefined;

/**
 * @member {Number} cp
 */
MarketQuoteSymbolLtpV3.prototype.cp = undefined;
