// upstox-js-sdk/dist/model/PlaceOrderV3Response.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PlaceOrderV3Response = void 0;
var _ApiClient = require("../ApiClient");
var _MultiOrderV3Data = require("./MultiOrderV3Data");
var _OrderMetadata = require("./OrderMetadata");
/*
 * OpenAPI definition
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 3.0.66
 *
 * Do not edit the class manually.
 *
 */

/**
 * The PlaceOrderV3Response model module.
 * @module model/PlaceOrderV3Response
 * @version v0
 */
class PlaceOrderV3Response {
  /**
   * Constructs a new <code>PlaceOrderV3Response</code>.
   * @alias module:model/PlaceOrderV3Response
   * @class
   */
  constructor() {}

  /**
   * Constructs a <code>PlaceOrderV3Response</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/PlaceOrderV3Response} obj Optional instance to populate.
   * @return {module:model/PlaceOrderV3Response} The populated <code>PlaceOrderV3Response</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new PlaceOrderV3Response();
      if (data.hasOwnProperty('status')) obj.status = _ApiClient.ApiClient.convertToType(data['status'], 'String');
      if (data.hasOwnProperty('data')) obj.data = _MultiOrderV3Data.MultiOrderV3Data.constructFromObject(data['data']);
      if (data.hasOwnProperty('metadata')) obj.metadata = _OrderMetadata.OrderMetadata.constructFromObject(data['metadata']);
    }
    return obj;
  }
}

/**
 * Allowed values for the <code>status</code> property.
 * @enum {String}
 * @readonly
 */
exports.PlaceOrderV3Response = PlaceOrderV3Response;
PlaceOrderV3Response.StatusEnum = {
  /**
   * value: "success"
   * @const
   */
  success: "success",
  /**
   * value: "error"
   * @const
   */
  error: "error",
  /**
   * value: "partial_success"
   * @const
   */
  partialSuccess: "partial_success"
};
/**
 * @member {module:model/PlaceOrderV3Response.StatusEnum} status
 */
PlaceOrderV3Response.prototype.status = undefined;

/**
 * @member {module:model/MultiOrderV3Data} data
 */
PlaceOrderV3Response.prototype.data = undefined;

/**
 * @member {module:model/OrderMetadata} metadata
 */
PlaceOrderV3Response.prototype.metadata = undefined;
