// upstox-js-sdk/dist/model/SearchInstrumentResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SearchInstrumentResponse = void 0;
var _ApiClient = require("../ApiClient");
var _SearchMetaData = require("./SearchMetaData");
var _InstrumentData = require("./InstrumentData");
/*
 * OpenAPI definition
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: v0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 3.0.66
 *
 * Do not edit the class manually.
 *
 */

/**
 * The SearchInstrumentResponse model module.
 * @module model/SearchInstrumentResponse
 * @version v0
 */
class SearchInstrumentResponse {
  /**
   * Constructs a new <code>SearchInstrumentResponse</code>.
   * @alias module:model/SearchInstrumentResponse
   * @class
   */
  constructor() {}

  /**
   * Constructs a <code>SearchInstrumentResponse</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/SearchInstrumentResponse} obj Optional instance to populate.
   * @return {module:model/SearchInstrumentResponse} The populated <code>SearchInstrumentResponse</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new SearchInstrumentResponse();
      if (data.hasOwnProperty('status')) obj.status = _ApiClient.ApiClient.convertToType(data['status'], 'String');
      if (data.hasOwnProperty('data')) obj.data = _ApiClient.ApiClient.convertToType(data['data'], [_InstrumentData.InstrumentData]);
      if (data.hasOwnProperty('meta_data')) obj.metaData = _ApiClient.ApiClient.convertToType(data['meta_data'], _SearchMetaData.SearchMetaData);
    }
    return obj;
  }
}

/**
 * @member {Object} status
 */
exports.SearchInstrumentResponse = SearchInstrumentResponse;
SearchInstrumentResponse.prototype.status = undefined;

/**
 * Response data for search instrument request
 * @member {Object} data
 */
SearchInstrumentResponse.prototype.data = undefined;

/**
 * Meta data for search instrument response
 * @member {module:model/SearchMetaData} metaData
 */
SearchInstrumentResponse.prototype.metaData = undefined;
